support/builder/build_dep_populate.sh

```diff
@@ -0,0 +1,463 @@
+#!/bin/bash
+
+set -euo pipefail
+
+aggMsg() {
+    echo ""
+    echo "Aggregating remotely stored artifacts."
+    echo "======================================"
+    echo ""
+
+}
+
+configMsg(){
+    echo ""
+    echo "Configuring environment for dep ingestion."
+    echo "======================================"
+    echo ""
+}
+
+keyMsg() {
+    echo ""
+    echo "Generating S3 Object Keys."
+    echo "======================================"
+    echo ""
+}
+
+uploadMsg() {
+    echo ""
+    echo "Uploading hartfiles."
+    echo "======================================"
+}
+
+getLatest() {
+   curl -s "${bldr_url}"/v1/depot/channels/core/stable/pkgs/"${1}"/latest | jq -r '.ident | [.origin,.name,.version,.release] | join("/")'
+}
+
+getHarts() {
+    # Traverses Core path in S3 bucket and
+    # creates an array from each package
+    # artifact that exists in that path
+    origin="core"
+    core_dirs=()
+    if [ "${s3type}" == 'minio' ]; then
+        # shellcheck disable=SC2207
+        core_dirs=( $(aws --endpoint-url "${AWS_REGION}" s3 ls "s3://${S3_BUCKET}/${origin}/") )
+    else
+        # shellcheck disable=SC2207
+        core_dirs=( $(aws --region "${AWS_REGION}" s3 ls "s3://${S3_BUCKET}/${origin}/") )
+    fi
+
+    for dir in "${core_dirs[@]}"; do
+        if [[ "${dir}" != "PRE" ]]; then
+            # shellcheck disable=SC2207
+            artifacts+=( $(curl -s "${bldr_url}/v1/depot/channels/core/stable/pkgs/${dir}latest" | jq -r '.ident | [.origin,.name,.version,.release] | join("/")') )
+        fi;
+    done
+    echo ""
+    echo "Downloading latest artifacts"
+    echo "This may take a moment"
+    echo ""
+
+    for hart in "${artifacts[@]}"; do
+        downloadHarts "${AWS_REGION}" "${S3_BUCKET}" "${hart}"
+    done
+}
+
+downloadHarts() {
+    download_path="/hab/tmp/harts/"
+    mkdir -p "${download_path}"
+
+    if [ "${s3type}" == 'minio' ]; then
+      aws --endpoint-url "${1}" s3 cp --recursive "s3://${2}/${3}" "${download_path}"
+    else
+      aws --region "${1}" s3 cp --recursive "s3://${2}/${3}" "${download_path}"
+    fi
+}
+
+checkBucket() {
+    region="${1}"
+    bucket="${2}"
+    aws --endpoint-url "${region}" s3api list-objects --bucket "${bucket}"
+}
+
+installDeps() {
+    PDEPS=("core/aws-cli" "core/jq-static" "habitat/s3-bulk-uploader")
+
+    configMsg
+
+    for bin in "${!PDEPS[@]}"; do
+        HAB_AUTH_TOKEN="" hab pkg install -b "${PDEPS[$bin]}"
+    done
+}
+
+setBucket() {
+    # Configure the bucket name to use for the upload
+    sgroups=("default" "dev" "prod" "acceptance" "live" "blue" "green")
+    for i in "${sgroups[@]}"; do
```

Comment:
Since you don't use `sgroups` array anywhere else, this could be
```bash
for i in default dev prod acceptance live blue green; do
```
Which I find a bit simpler, but it's really just a personal preference